opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go

```diff
@@ -0,0 +1,373 @@
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					}},
+			},
+			"topic_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"notification_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"notification_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"project_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"create_time": {
+				Type:     schema.TypeFloat,
+				Computed: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceCTSOperationsV3(d *schema.ResourceData) []keyevent.Operations {
+	operations := d.Get("operations").([]interface{})
+	if len(operations) == 0 {
+		return nil
+	}
+	var keyEvents []keyevent.Operations
+
+	for _, v := range operations {
+		operation := v.(map[string]interface{})
+
+		var refinedTraces []string
+
+		for _, s := range operation["trace_names"].([]interface{}) {
+			refinedTraces = append(refinedTraces, s.(string))
+		}
+
+		keyEvent := keyevent.Operations{
+			ServiceType:  operation["service_type"].(string),
+			ResourceType: operation["resource_type"].(string),
+			TraceNames:   refinedTraces,
+		}
+		keyEvents = append(keyEvents, keyEvent)
+	}
+
+	return keyEvents
+}
+
+func resourceCTSUserListV3(d *schema.ResourceData) []keyevent.NotificationUsers {
+	userList := d.Get("notify_user_list").([]interface{})
+	if len(userList) == 0 {
+		return nil
+	}
+
+	var usersOpts []keyevent.NotificationUsers
+
+	for _, v := range userList {
+		user := v.(map[string]interface{})
+
+		var refinedUserList []string
+
+		for _, s := range user["user_list"].([]interface{}) {
+			refinedUserList = append(refinedUserList, s.(string))
+		}
+
+		userOpts := keyevent.NotificationUsers{
+			UserGroup: user["user_group"].(string),
+			UserList:  refinedUserList,
+		}
+		usersOpts = append(usersOpts, userOpts)
+	}
+
+	return usersOpts
+}
+
+func resourceCTSEventNotificationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.CtsV3Client(config.GetProjectName(d))
+	if err != nil {
+		return fmterr.Errorf(clientError, err)
+	}
+
+	createOpts := keyevent.CreateNotificationOpts{
+		NotificationName: d.Get("notification_name").(string),
+		OperationType:    d.Get("operation_type").(string),
+		Operations:       resourceCTSOperationsV3(d),
+		NotifyUserList:   resourceCTSUserListV3(d),
+		TopicId:          d.Get("topic_id").(string),
+	}
+
+	ctsN, err := keyevent.Create(client, createOpts)
+	if err != nil {
+		return fmterr.Errorf("error creating CTS event notification: %w", err)
+	}
+
+	d.SetId(buildEventNotificationID(ctsN.NotificationId, ctsN.NotificationName))
+
+	if status, ok := d.GetOk("status"); ok && createOpts.TopicId != "" {
```

Comment:
you really need this if, impossible to use one clause?

Reply:
fixed

Reply:
It's possible but imo this way is more understandable than something like this:
`if status, ok := d.GetOk("status"); ok && createOpts.TopicId != "" && status.(string) == "disabled" {`
What do you think?